metatheory/Main.lagda

```diff
@@ -50,20 +54,58 @@ postulate
 {-# FOREIGN GHC import qualified Data.Text as T #-}
 {-# COMPILE GHC putStrLn = Text.putStrLn #-}
 
+-- IO Stuff
+
 postulate
-  return : ∀ {a} {A : Set a} → A → IO A
-  _>>=_  : ∀ {a b} {A : Set a} {B : Set b} → IO A → (A → IO B) → IO B
+  returnIO : ∀ {a} {A : Set a} → A → IO A
+  _>>=IO_  : ∀ {a b} {A : Set a} {B : Set b} → IO A → (A → IO B) → IO B
+  imap : ∀{A B : Set} → (A → B) → IO A → IO B
 
-{-# COMPILE GHC return = \_ _ -> return    #-}
-{-# COMPILE GHC _>>=_  = \_ _ _ _ -> (>>=) #-}
+instance
+  IOMonad : Monad IO
+  IOMonad = record { return = returnIO ; _>>=_ = _>>=IO_ }
 
-_>>_  : ∀ {a b} {A : Set a} {B : Set b} → IO A → IO B → IO B
+_>>_  : {A : Set} {B : Set} → IO A → IO B → IO B
```

Comment:
No universe polymorphism here, but not in the bind?

Reply:
I removed the universe polymorphism from bind too, my preference is not only add it when needed.